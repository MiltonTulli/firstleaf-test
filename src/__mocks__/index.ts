export const countries = [
  {
    name: 'Afghanistan',
    alpha2Code: 'AF',
    alpha3Code: 'AFG',
    capital: 'Kabul',
    region: 'Asia',
    subregion: 'Southern Asia',
    population: 27657145,
    latlng: [33, 65],
    demonym: 'Afghan',
    area: 652230,
    gini: 27.8,
    timezones: ['UTC+04:30'],
    borders: ['IRN', 'PAK', 'TKM', 'UZB', 'TJK', 'CHN'],
    nativeName: 'افغانستان',
    currencies: [
      {
        code: 'AFN',
        name: 'Afghan afghani',
        symbol: '؋'
      }
    ],
    languages: [
      {
        iso639_1: 'ps',
        iso639_2: 'pus',
        name: 'Pashto',
        nativeName: 'پښتو'
      },
      {
        iso639_1: 'uz',
        iso639_2: 'uzb',
        name: 'Uzbek',
        nativeName: 'Oʻzbek'
      },
      {
        iso639_1: 'tk',
        iso639_2: 'tuk',
        name: 'Turkmen',
        nativeName: 'Türkmen'
      }
    ],
    flag: 'https://restcountries.eu/data/afg.svg'
  },
  {
    name: 'Åland Islands',
    alpha2Code: 'AX',
    alpha3Code: 'ALA',
    capital: 'Mariehamn',
    region: 'Europe',
    subregion: 'Northern Europe',
    population: 28875,
    latlng: [60.116667, 19.9],
    demonym: 'Ålandish',
    area: 1580,
    gini: null,
    timezones: ['UTC+02:00'],
    borders: [],
    nativeName: 'Åland',
    currencies: [
      {
        code: 'EUR',
        name: 'Euro',
        symbol: '€'
      }
    ],
    languages: [
      {
        iso639_1: 'sv',
        iso639_2: 'swe',
        name: 'Swedish',
        nativeName: 'svenska'
      }
    ],
    flag: 'https://restcountries.eu/data/ala.svg'
  },
  {
    name: 'Albania',
    alpha2Code: 'AL',
    alpha3Code: 'ALB',
    capital: 'Tirana',
    region: 'Europe',
    subregion: 'Southern Europe',
    population: 2886026,
    latlng: [41, 20],
    demonym: 'Albanian',
    area: 28748,
    gini: 34.5,
    timezones: ['UTC+01:00'],
    borders: ['MNE', 'GRC', 'MKD', 'KOS'],
    nativeName: 'Shqipëria',
    currencies: [
      {
        code: 'ALL',
        name: 'Albanian lek',
        symbol: 'L'
      }
    ],
    languages: [
      {
        iso639_1: 'sq',
        iso639_2: 'sqi',
        name: 'Albanian',
        nativeName: 'Shqip'
      }
    ],
    flag: 'https://restcountries.eu/data/alb.svg'
  },
  {
    name: 'Algeria',
    alpha2Code: 'DZ',
    alpha3Code: 'DZA',
    capital: 'Algiers',
    region: 'Africa',
    subregion: 'Northern Africa',
    population: 40400000,
    latlng: [28, 3],
    demonym: 'Algerian',
    area: 2381741,
    gini: 35.3,
    timezones: ['UTC+01:00'],
    borders: ['TUN', 'LBY', 'NER', 'ESH', 'MRT', 'MLI', 'MAR'],
    nativeName: 'الجزائر',
    currencies: [
      {
        code: 'DZD',
        name: 'Algerian dinar',
        symbol: 'د.ج'
      }
    ],
    languages: [
      {
        iso639_1: 'ar',
        iso639_2: 'ara',
        name: 'Arabic',
        nativeName: 'العربية'
      }
    ],
    flag: 'https://restcountries.eu/data/dza.svg'
  }
];
