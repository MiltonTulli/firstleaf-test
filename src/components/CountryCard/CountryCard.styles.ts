import { makeStyles } from '@material-ui/core/styles';
import { Theme } from '@material-ui/core';

type StyleProps = { backgroundUrl: string };

export default makeStyles<Theme, StyleProps>(theme => ({
  root: {
    margin: '4px 0px',
    padding: '0 4px',
    [theme.breakpoints.down('xs')]: {
      width: 165,
      height: 'fit-content'
    }
  },
  contentWrapper: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'column'
  },
  flag: {
    height: 50,
    boxShadow: theme.shadows[10],
    margin: '0 auto 16px',
    borderRadius: 10
  }
}));
