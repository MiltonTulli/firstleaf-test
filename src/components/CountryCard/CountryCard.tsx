import React from 'react';
import { Card, CardActionArea, CardContent, Typography } from '@material-ui/core';
import { Country } from '../../interfaces';
import useStyles from './CountryCard.styles';

interface ICountryCardProps {
  country: Country;
  handleClick: (country: Country) => void;
}

export const CountryCard: React.FC<ICountryCardProps> = (props: ICountryCardProps) => {
  const { country, handleClick } = props;
  const classes = useStyles({ backgroundUrl: country.flag });

  return (
    <div data-testid="card" onClick={() => handleClick(country)} className={classes.root}>
      <Card className={classes.root} elevation={10}>
        <CardActionArea>
          <CardContent>
            <div className={classes.contentWrapper}>
              <img src={country.flag} className={classes.flag} alt={`${country.name} flag`} />
              <Typography gutterBottom variant="body1" component="h1" align="center">
                {country.name}
              </Typography>
            </div>
          </CardContent>
        </CardActionArea>
      </Card>
    </div>
  );
};
