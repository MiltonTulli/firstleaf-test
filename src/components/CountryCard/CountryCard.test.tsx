import React from 'react';
import { shallow } from 'enzyme';
import { CountryCard } from './CountryCard';
import { countries } from '../../__mocks__';

describe('<CountryCard>', () => {
  let props;

  beforeEach(() => {
    props = {
      country: countries[0],
      handleClick: jest.fn()
    };
  });

  it('Should match snapshot', () => {
    const wrapper = shallow(<CountryCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call handleClick', async () => {
    const wrapper = shallow(<CountryCard {...props} />);
    const card = wrapper.find('[data-testid="card"]');
    card.simulate('click');
    expect(props.handleClick).toHaveBeenLastCalledWith(props.country);
  });

  it('Should show country name', async () => {
    const wrapper = shallow(<CountryCard {...props} />);
    expect(wrapper.text()).toContain(props.country.name);
  });
});
