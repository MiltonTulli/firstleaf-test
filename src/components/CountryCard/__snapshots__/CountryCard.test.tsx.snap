// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountryCard> Should match snapshot 1`] = `
<div
  className="makeStyles-root-1"
  data-testid="card"
  onClick={[Function]}
>
  <WithStyles(ForwardRef(Card))
    className="makeStyles-root-1"
    elevation={10}
  >
    <WithStyles(ForwardRef(CardActionArea))>
      <WithStyles(ForwardRef(CardContent))>
        <div
          className="makeStyles-contentWrapper-2"
        >
          <img
            alt="Afghanistan flag"
            className="makeStyles-flag-3"
            src="https://restcountries.eu/data/afg.svg"
          />
          <WithStyles(ForwardRef(Typography))
            align="center"
            component="h1"
            gutterBottom={true}
            variant="body1"
          >
            Afghanistan
          </WithStyles(ForwardRef(Typography))>
        </div>
      </WithStyles(ForwardRef(CardContent))>
    </WithStyles(ForwardRef(CardActionArea))>
  </WithStyles(ForwardRef(Card))>
</div>
`;
